### ISSUE TITLE
[PROBLEM] javax.money.spi.MonetaryCurrenciesSingletonSpi: module org.javamoney.moneta does not declare `uses`

### ISSUE THREAD

**Issue by AndreiOprisan94** [JavaMoney/jsr354-ri#300]
Hello, 

I am trying to use the Java Money Api. My pom.xml declares the dependency like this:

```       
             <dependency>
                    <groupId>org.javamoney</groupId>
                    <artifactId>moneta</artifactId>
                     <version>1.3</version>
                    <scope>runtime</scope>
            </dependency>
```

I am trying to create a new `MonetaryAmount` with the following call chain `Monetary.getDefaultAmountFactory().setCurrency(usd).setNumber(500).create()`

At this point it fails with the following stack trace:

```Exception in thread "main" java.util.ServiceConfigurationError: javax.money.spi.MonetaryCurrenciesSingletonSpi: module org.javamoney.moneta does not declare \`uses\`
	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:588)
	at java.base/java.util.ServiceLoader.checkCaller(ServiceLoader.java:574)
	at java.base/java.util.ServiceLoader.<init>(ServiceLoader.java:503)
	at java.base/java.util.ServiceLoader.load(ServiceLoader.java:1691)
	at org.javamoney.moneta/org.javamoney.moneta.internal.PriorityAwareServiceProvider.loadServices(PriorityAwareServiceProvider.java:98)
	at org.javamoney.moneta/org.javamoney.moneta.internal.PriorityAwareServiceProvider.getServices(PriorityAwareServiceProvider.java:65)
	at java.money/javax.money.spi.Bootstrap.getService(Bootstrap.java:113)
	at java.money/javax.money.Monetary.MONETARY_CURRENCIES_SINGLETON_SPI(Monetary.java:42)
	at java.money/javax.money.Monetary.getCurrency(Monetary.java:382)```

I am trying to run the program from Intellij Version 2018.3.4. With Java Version 11 on OpenJDK.

Any advice would be kindly appreciated. Thank you!

**Comment by Daniel-Dos** [comment]
Hi @AndreiOprisan94 ,

you have an exemple in your github for I test ?

I made a test with Eclipse and openJDK 11 plus moneta 1.3 and for me not presented this error . 

![test](https://user-images.githubusercontent.com/8139890/53712650-84405a00-3e26-11e9-8433-3b6e1dc76a86.png)

**Comment by Daniel-Dos** [comment]
I think what dependency correct is this :

```
<dependencies>
        <!-- https://mvnrepository.com/artifact/org.javamoney/moneta -->
        <dependency>
            <groupId>org.javamoney</groupId>
            <artifactId>moneta</artifactId>
            <version>1.3</version>
            <type>pom</type>
        </dependency>
    </dependencies>
```

**Comment by AndreiOprisan94** [comment]
Hi,

It does not work.

Please see the sample project here: https://github.com/AndreiOprisan94/TestJavaMoney

Try and run class `MainApp` from module `orders-service` and see if it works. To me it does not. It fails with the error I mentioned.

I decompiled using Intellij the `module-info.class` file of the `moneta` jar and it does NOT contain the statement `uses javax.money.spi.MonetaryCurrenciesSingletonSpi`. 

Thanks!

**Comment by Daniel-Dos** [comment]
Hi, 

I see this here.

thanks

**Comment by Daniel-Dos** [comment]
Hi @AndreiOprisan94 ,

I ran your project in my eclipse and it ran without any problems.

![teste](https://user-images.githubusercontent.com/8139890/53777491-56b3e900-3ed8-11e9-95f7-2d7ffd30beee.png)


![teste2](https://user-images.githubusercontent.com/8139890/53777519-68958c00-3ed8-11e9-8629-56482e4b398d.png)

**Comment by stokito** [comment]
looks like duplicate of #219

**Comment by Daniel-Dos** [comment]
@AndreiOprisan94 please try delete the folder org.javamoney e javax.money in your folder .m2 and test again .

**Comment by nemphys** [comment]
Same problem here (same environment, building with gradle).

**Comment by nemphys** [comment]
It seems that this problem has been solved somewhere around Sept. 2018, but the 1.3 (latest) release dates back to July 2018 (!), so the fix is not included.
Guess we will have to build from source...

**Comment by AndreiOprisan94** [comment]
@Daniel-Dos I have deleted the whole local `.m2` folder. And did a new `mvn clean package` on the root project. Furthermore I get an error when I run this command in the CLI. The error is saying:

`[ERROR] Failed to execute goal on project orders-service: Could not resolve dependencies for project com.pharmaplus:orders-service:jar:1.0-SNAPSHOT: Could not find artifact org.javamoney:moneta:jar:1.3 in central (https://repo.maven.apache.org/maven2) -> [Help 1]`

It seems that the artefact is no longer on the repo :(. Following link gives the HTTP 404 status code https://repo.maven.apache.org/maven2/org/javamoney/moneta/1.3/moneta-1.3.jar.

**Comment by stokito** [comment]
@AndreiOprisan94 
That's because you should change `<scope>` from `runtime` to `pom`:

             <dependency>
                    <groupId>org.javamoney</groupId>
                    <artifactId>moneta</artifactId>
                     <version>1.3</version>
                    <scope>pom</scope>
            </dependency>

But please note that Maven had a bug with such dependencies so you may need to update Maven to latest.

**Comment by keilw** [comment]
So can we close this?

**Comment by Daniel-Dos** [comment]
@AndreiOprisan94 ,

I too made test in terminal and startup normal: 

![jdk11](https://user-images.githubusercontent.com/8139890/54080642-569a5b80-42d3-11e9-9a56-2439de9dacdc.png)

**Comment by Daniel-Dos** [comment]
@nemphys 

I not made rebuild of source .

**Comment by AndreiOprisan94** [comment]
Sorry. But I have changed the dependency from `runtime` to `pom` like @stokito mentioned, still 404 problem. I have deleted the `.m2` directory from my home directory and rebuild. Still the same problem.

Software I am using:

* macOS Sierra (Version 10.12.6)
* OpenJDK 11 Runtime Environment 18.9 (build 11+28)
* Maven 3.6.0
* Intellij 2018.3.4.

If I have time. I will build moneta from source file. And maybe ask someone to try the project on hes own machine.

**Comment by Daniel-Dos** [comment]
Hi @AndreiOprisan94 

the correct is : 

```
<dependency>
            <groupId>org.javamoney</groupId>
            <artifactId>moneta</artifactId>
            <version>${moneta.version}</version>
            <type>pom</type>
        </dependency>
```

**Comment by AndreiOprisan94** [comment]
Now it throws `NullPointerException` :(

**Comment by AndreiOprisan94** [comment]
On `mvn clean package`

**Comment by keilw** [comment]
Did you correctly set `${moneta.version}` in your POM?

**Comment by AndreiOprisan94** [comment]
yes. to version 1.3

**Comment by nemphys** [comment]
In order to get this to work, I had to clone the repo and build local jars with the following additions to module-info.java:

    uses javax.money.spi.CurrencyProviderSpi;
    uses javax.money.spi.MonetaryCurrenciesSingletonSpi;
    uses javax.money.spi.MonetaryAmountFactoryProviderSpi;
    uses javax.money.spi.MonetaryAmountFormatProviderSpi;
    uses javax.money.spi.MonetaryAmountsSingletonQuerySpi;
    uses javax.money.spi.MonetaryAmountsSingletonSpi;
    uses javax.money.spi.MonetaryConversionsSingletonSpi;
    uses javax.money.spi.MonetaryFormatsSingletonSpi;
    uses javax.money.spi.MonetaryRoundingsSingletonSpi;
    uses javax.money.spi.RoundingProviderSpi;
    uses javax.money.spi.ServiceProvider;
    uses javax.money.convert.ExchangeRateProvider;
    uses javax.money.convert.ExchangeRateProviderSupplier;
    uses javax.money.format.MonetaryAmountFormat;

After including the jars in my project, everything works fine.

**Comment by keilw** [comment]
Could someone, ideally from the EG (like @Daniel-Dos) propose a PR based on that?

**Comment by stokito** [comment]
Sorry, I'm busy on this week so @Daniel-Dos if you can please tell us

**Comment by Daniel-Dos** [comment]
hello 
can i do yes, i will look at this, as presented above

**Comment by Daniel-Dos** [comment]
fix in #302 
please close this issue .
